src/Compilers/CSharp/Portable/BoundTree/Constructors.cs

```diff
@@ -725,4 +725,24 @@ public BoundIncrementOperator Update(UnaryOperatorKind operatorKind, BoundExpres
             return Update(operatorKind, operand, methodOpt, constrainedToTypeOpt, operandConversion, resultConversion, resultKind, this.OriginalUserDefinedOperatorsOpt, type);
         }
     }
+
+    internal partial class BoundPointerIndirectionOperator
+    {
+        public BoundPointerIndirectionOperator(SyntaxNode syntax, BoundExpression operand, TypeSymbol type, bool hasErrors = false)
```

Comment:
>BoundPointerIndirectionOperator [](http://example.com/codeflow?start=15&length=31)

I think adding constructors like that is error prone, easy to miss the fact that a "non-default" value should be used. It also doesn't look like constructors are used very often, not that many changes are avoided. Same comment for the other bound node.